packages/e2e-tests/src/specs/editor/fontCheck/fontCheck.js

```diff
@@ -0,0 +1,156 @@
+  await page.keyboard.press('ArrowDown');
+  await page.keyboard.press('Enter');
+  await expect(page).toClick('button', { text: 'Replace font' });
+  await expect(page).toClick('[data-testid="textFrame"]');
+  await expect(page).toClick('button[aria-label="Font family"]');
+};
+
+const replaceFontUsingDefault = async () => {
+  await page.waitForSelector('[role="dialog"]');
+  await expect(page).toClick('button', { text: 'Open anyway' });
+  await expect(page).toClick('[data-testid="textFrame"]');
+  await expect(page).toClick('button[aria-label="Font family"]');
+};
+
+const storyWithFontCheckDialog = async (
+  title,
+  customFont = 'Open Sans Condensed Light'
+) => {
+  // take steps needed to get Missing Font dialog to show
+  await addCustomFont();
+  await createStoryWithTitle(title);
+  await updateFont(customFont);
+  await publishStory();
+  await removeCustomFont();
+  await visitStoryWithTitle(title);
+  await page.waitForSelector('[data-testid="textFrame"]');
+  await page.waitForSelector('[role="dialog"]');
+};
+
+describe('Font Check', () => {
+  describe('Font Check Dialog', () => {
+    withExperimentalFeatures(['customFonts']);
+    withExperimentalFeatures(['notifyDeletedFonts']);
+
+    it('should show dialog & replace font with default font', async () => {
+      const title = 'Test replace missing font with (default) Roboto';
+      const replacementFont = 'Roboto';
+      await storyWithFontCheckDialog(title);
+      await replaceFontUsingDefault(replacementFont);
+      const selectedText = await page.$eval(
+        'div[role="dialog"] li[aria-selected="true"]',
+        (el) => el.textContent
+      );
+      await expect(selectedText).toBe(replacementFont);
+    });
+
+    it('should show dialog & replace it with selected font', async () => {
+      const title = 'Test replace missing font with Bungee';
+      const replacementFont = 'Bungee';
+      await storyWithFontCheckDialog(title);
+      await replaceFontWithFontPicker(replacementFont);
+      const selectedText = await page.$eval(
+        'div[role="dialog"] li[aria-selected="true"]',
+        (el) => el.textContent
+      );
+      await expect(selectedText).toBe(replacementFont);
+    });
+
+    it('should show dialog & visit settings page', async () => {
+      const title = 'Test visit to settings from dialog';
+      await storyWithFontCheckDialog(title);
+      await Promise.all([
+        expect(page).toClick('[role="dialog"] a', { text: 'Settings' }),
+        page.waitForNavigation(),
+      ]);
+      await takeSnapshot(page, 'back-to-settings');
+      await expect(page).toMatchElement('h2', { text: 'Settings' });
+    });
+
+    it('should show dialog & visit dashboard page', async () => {
+      const title = 'Test back to dashboard from dialog';
+      await storyWithFontCheckDialog(title);
+      await Promise.all([
+        expect(page).toClick('a', { text: 'Back to dashboard' }),
+        page.waitForNavigation(),
+      ]);
```

Comment:
I know we have used this snippet elsewhere. But it has been problematic in the past. If `waitForNavigation` runs first, before the click, the it returns resolved. This snippet has failed in the past.

Reply:
This seems to be the most solid solution from all the test runs I did -- without adding the Promise.all ... I was hitting white screens etc... even with the network idle flag.

Have another solution ?

Reply:
Let's leave it for now then. But flagging for the future.